### ISSUE TITLE
Feature request: Ability to truncate input files

### ISSUE THREAD

**Issue by ant0nsc** [Microsoft/CNTK#312]
I've got a data set with several dozens of millions of images. I'm now gradually trying to ramp up the training set, to figure out which architectures work well. At present, this means I have map files with 1k images, 100k, 1M, etc. I think it would be helpful if I could have an option to do that in the CNTK config - work with the same mapfile, but have an option "truncate = 1000" to only use the first 1000 examples from the map file. I had a look through the reader code, and did not find any way how I can do that at present.

**Comment by yzhang87** [comment]
Do you mean only use 1000 examples? I think just set epochSize=1000, it will only use the first 1000 examples.

**Comment by frankseide** [comment]
This depends on the readers. Some readers also allow Perl-like pipe syntax, where you could run the corpus description through `head` command. Here is an actual example I used:
```
embeddingsPath = "$DataDir$/voc.8k.emb.txt"    # word space embedding-vector, one line per word
embeddings = "gawk '{$1 = EMPTY; print}' $embeddingsPath$ |"
mappings   = "gawk '{print $1}'          $embeddingsPath$ |"
```
Again, not all readers support this at present.

**Comment by ant0nsc** [comment]
I tried the suggestion by @yzhang87 but it failed with an access violation. Will send a bug report.

**Comment by ant0nsc** [comment]
Seems my GPU had a bad day. Today, using `epochSize` works just fine. Thanks to @yzhang87 for pointing me in the right direction, closing the issue.

**Comment by ant0nsc** [comment]
Upon further investigation, it seems the problem was caused by a combination of 1) other jobs running on the GPU, and 2) old CUDA kit. With latest CNTK with CUDA 7.5, I see no exceptions anymore.

**Comment by ant0nsc** [comment]
I'm putting this back on the wishlist. I just ran a setup where I am using a `cvReader` for learning rate control, and I found no way to truncate my input other than to mess with my input file.
Rather than changing every place where input data is consumed, I feel it would be better if readers had a consistent support for that:

- Skipping N lines at the beginning of the file
- Truncating its output to at most M lines

With those two parameters, I would get a config-driven way of extracting any (contiguous) subset of lines from any input.
Is that feasible?

**Comment by wolfma61** [comment]
Hi - not sure if this can be solved with the new reader design - please take a look

I'm closing this one, since the original topic has been solved. If you require additional feature, please open a new issue with a detailed requirement spec ...
